packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1247,4 +1247,315 @@ void main() {
     expect(find.text('C'), findsOneWidget);
     expect(find.text('D'), findsNothing);
   });
+
+  testWidgets('Custom expandIconBuilder - does not trigger ExpansionList expansionCallback', (WidgetTester tester) async {
+    final List<bool> _isExpanded = <bool>[true, false];
+    bool isTriggered = false;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SingleChildScrollView(
+          child: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState)  {
+              return ExpansionPanelList(
+                expansionCallback: (int index, bool isExpanded) {
+                  isTriggered = true;
+                  setState(() { _isExpanded[index] = !isExpanded; });
+                },
+                children: <ExpansionPanel>[
+                  ExpansionPanel(
+                    headerBuilder: (BuildContext context, bool isExpanded) {
+                      return const Text('Header');
+                    },
+                    expandIconBuilder: (BuildContext context, bool isExpanded) {
+                      return Checkbox(
+                        value: isExpanded,
+                        onChanged: (bool val) {
+                          setState(() { _isExpanded[0] = val; });
+                        },
+                      );
+                    },
+                    body: _isExpanded[0] ? const Text('B') : const Text('A'),
+                    isExpanded: _isExpanded[0],
+                  ),
+                  ExpansionPanel(
+                    headerBuilder: (BuildContext context, bool isExpanded) {
+                      return const Text('Header');
+                    },
+                    expandIconBuilder: (BuildContext context, bool isExpanded) {
+                      return Checkbox(
+                        value: isExpanded,
+                        onChanged: (bool val) {
+                          setState(() { _isExpanded[1] = val; });
+                        },
+                      );
+                    },
+                    body: _isExpanded[1] ? const Text('D') : const Text('C'),
+                    isExpanded: _isExpanded[1],
+                  ),
+                ],
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(Checkbox), findsNWidgets(2));
```

Comment:
It will be good to add a check to make sure checkbox is unchecked state before tap and a checked state after tap, essentially to test the isExpanded is correctly passed into expandIconBuilder

Reply:
Done